lib/internal/modules/esm/loader.js

```diff
@@ -802,15 +817,17 @@ class ESMLoader {
       hookErrIdentifier: '',
       hookIndex: chain.length - 1,
       hookName: 'resolve',
+      isChainAsync: false,
       shortCircuited: false,
     };
-
     const context = {
       conditions: DEFAULT_CONDITIONS,
       importAssertions,
       parentURL,
     };
-    const validate = (hookErrIdentifier, { 0: suppliedSpecifier, 1: ctx }) => {
+
+    const validate = (hookErrIdentifier, output) => {
+      const { 0: suppliedSpecifier, 1: ctx } = output;
```

Comment:
nit: we could keep that line as it was before
```suggestion
const validate = (hookErrIdentifier, { 0: suppliedSpecifier, 1: ctx }) => {
```